### PR TITLE
Continuing from #1317 and its PRs for pistache server string enum code generation

### PR DESCRIPTION
**Pull request by mfyuce**
* A class that has an `anyOf` specification, in cpp side will have no members: instead it should have a member having the type `stringenumclassname_anyOf`
* Thus, Its `==` operator is not present or wrongly filled
* A string enum, should have a better usage, hence the `setEnumValue`
* this PR, is a brigde between `stringenumclassname_**anyOf**` and `stringenumclassname`
* `anyOf` specification is not just about `Enums`, so a better handling is needed from mustache templates, hence new template model parameter `isStringEnumContainer`

### PR checklist
 
- [ X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@ravinikam (2017/07) @stkrwork (2017/07) @etherealjoy (2018/02) @MartinDelille (2018/03) @muttleyxd (2019/08)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/cpp-pistache-server/model-source.mustache [hunk 1]
```diff
@@ -60,21 +60,22 @@ bool {{classname}}::validate(std::stringstream& msg, const std::string& pathPref
...
         {{> model-validation-body }}
     }
-    {{/hasValidation}}{{/isArray}}
-    {{/vars}}
-    {{/isEnum}}
-
+    {{/hasValidation}}{{/isArray}}{{/vars}}{{/isEnum}}{{#isStringEnumContainer}}{{#anyOf}}{{#-first}}
+    if (!m_value.validate(msg))
+    {
+        success = false;;
```
Comment (muttleyxd) [inline modules/openapi-generator/src/main/resources/cpp-pistache-server/model-source.mustache:66]: double semicolon

**Review by muttleyxd — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/cpp-pistache-server/model-source.mustache (hunk 1 above)
Comment (mfyuce) [reply modules/openapi-generator/src/main/resources/cpp-pistache-server/model-source.mustache]: Thanks. It is fixed.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 2]
```diff
@@ -2634,6 +2634,14 @@ public int compare(CodegenProperty one, CodegenProperty another) {
                 postProcessModelProperty(m, prop);
             }
         }
+        if(!m.isEnum
+                && m.anyOf.size()>1
+                && m.anyOf.contains("std::string")
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:2639]: I think `std::string` is C++ only.

What about adding `x-is-string-enum-container` instead in the postProcessModel operation in the C++ pistache server generator?

Comment (mfyuce) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: Thanks.

- Removed "isStringEnumContainer" and its setter from "DefaultCodeGen"
- added the setter to "CppPistacheServerCodegen.java" making it add a new entry to "vendorExtensions" as "x-is-string-enum-container"

Now it is pistache cpp server only. Making it global to all servers...; you may guide me..

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: If other generators need the same, they can reuse the same code block in the postProcessModel method.

**Pull request merged**